test/DynamoCoreTests/Graph/Nodes/NodeModelTests.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.ObjectModel;
+using Dynamo.Graph.Nodes;
+using NUnit.Framework;
+using ProtoCore.AST;
+using Revit.Elements;
+
+namespace Dynamo.Tests
+{
+    /// <summary>
+    /// Class containing tests for the NodeModel class
+    /// </summary>
+    [TestFixture]
+    class NodeModelTests : DynamoModelTestBase
+    {
+        NodeModel testNodeModel;
+
+        [SetUp]
+        public void Init()
+        {
+            CurrentDynamoModel.NodeFactory.CreateNodeFromTypeName("CoreNodeModels.Input.DoubleInput", out testNodeModel);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void CreationNameTest()
+        {
+            testNodeModel.Category = "Category";
+
+            var result = testNodeModel.ConstructDictionaryLinkFromLibrary(CurrentDynamoModel.LibraryServices);
+            Assert.AreEqual("http://dictionary.dynamobim.com/2/#/Category/Action/Number", result);
+        }
+        
+        [Test]
+        [Category("UnitTests")]
+        public void InPortsOutPortsTest()
+        {
+            var input = new PortModel(PortType.Input, testNodeModel, new PortData("input A", "This is input A."));
+            var inPorts = new ObservableCollection<PortModel>() { input };
+            testNodeModel.InPorts = inPorts;
+
+            Assert.Contains(input, testNodeModel.InPorts);
+
+            var output = new PortModel(PortType.Output, testNodeModel, new PortData("output A", "This is output A."));
+            var outPorts = new ObservableCollection<PortModel>() { output };
+            testNodeModel.OutPorts = outPorts;
+
+            Assert.Contains(output, testNodeModel.OutPorts);
```

Comment:
This doesn't look like a very useful test.
All you're doing is adding something to a collection and then checking if the collection contains that thing :)

Reply:
What I'm trying to do here is to cover the setter for the InPorts and OutPorts properties. 

Since these are both collections, naturally every manipulation of them in the project is either adding or removing elements, but never replacing the entire collection by a new one like I'm doing there.

How do you think we could write a more useful test for this scenario?

Reply:
It looks like there is no place in the code base where these properties are being assigned directly, which is why the setter isn't covered. Nor do I think does the `RaisePropertyChangedEvent` do anything. I think this is a candidate for dead code removal?

Reply:
That sounds like a better approach. Should I go ahead and remove the setters for these two properties together with the test?

Reply:
I don't think you can remove them right away as they are public setters.

Reply:
Should I remove that test?

Reply:
I think you can apply the `IsObsolete` custom attribute to the setter with a message saying it will be deprecated in Dynamo 3.0. Yes, I also feel the test is not necessary.

Reply:
Alright! I'll add the attribute and remove the test.